scripts/sanity_chk/sanitylib.py

```diff
@@ -3229,6 +3232,45 @@ def csv_report(self, filename):
                     rowdict["rom_size"] = rom_size
                 cw.writerow(rowdict)
 
+    def set_default(self, obj):
+        ''' JSON doesn't recognize set type so need to converted to list '''
+        if isinstance(obj,set):
+            return list (obj)
+        raise TypeError
+
+    def json_report(self, filename):
+        rowdict = {}
+        rowdict["environment"] = {"OS": os.name,
+                                    "Toolchain": self.get_toolchain()
+                                    }
+        rowdict["testspec"] = []
+        for instance in self.instances.values():
+            rowdict["testspec"].append({"test": instance.testcase.name,
+                                        "arch": instance.platform.arch,
+                                        "platform": instance.platform.name,
+                                        "handler": instance.platform.simulation,
+                                        "status": instance.status,
+                                        "extra_args": instance.testcase.extra_args,
+                                        "extra_configs": instance.testcase.extra_configs,
+                                        "harness": instance.testcase.harness,
+                                        "harness_config": instance.testcase.harness_config,
+                                        "type": instance.testcase.type,
+                                        "tags": instance.testcase.tags
+                                        })
+            if instance.status not in ["failed", "timeout"]:
+                if instance.handler:
+                    rowdict["testspec"].append({"handler_time": (instance.metrics.get           ("handler_time", 0))})
+                rowdict["testspec"].append({"ram_size": instance.metrics.get("ram_size", 0),
+                                            "rom_size": instance.metrics.get("rom_size",0)
+                                            })
+        rowdict["apply_filters"] = {"discards": str(self.apply_filters())}
```

Comment:
this is just dumping a dictionary into json, the data from discards should be per instance in this case, looks like this in json which is not ok:

```
{
   "apply_filters" : {
      "discards" : "{<TestCase samples/hello_world/sample.basic.helloworld on esp32>: 'Not supported by the toolchain', <TestCase samples/hello_world/sample.basic.helloworld on xt-sim>: 'Not supported by the toolchain', <TestCase samples/hello_world/sample.basic.helloworld on esp32>: 'Not supported by the toolchain', <TestCase samples/hello_world/sample.basic.helloworld on qemu_x86_nommu>: 'Excluded tags per platform (only_tags)', <TestCase samples/hello_world/sample.basic.helloworld on qemu_x86_coverage>: 'Excluded tags per platform (only_tags)', <TestCase samples/hello_world/sample.basic.helloworld on qemu_x86_xip>: 'Excluded tags per platform (only_tags)', <TestCase samples/hello_world/sample.basic.helloworld on nrf52_bsim>: 'Environment (BSIM_OUT_PATH) not satisfied', <TestCase samples/hello_world/sample.basic.helloworld on stm32f030_demo>: 'Not enough RAM',
```

Reply:
addressed it, now it adds per instance and details.